File: src/main/java/com.github.rogerbarton.intellijsv/SvFileType.java
package com.github.rogerbarton.intellijsv;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public class SvFileType extends LanguageFileType {
    public static final SvFileType INSTANCE = new SvFileType();

    private SvFileType() {
        super(SvLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "System Verilog File";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "System Verilog (HDL) File";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "sv";
    }

    @NotNull
    @Override
    public Icon getIcon() {
        return SvIcons.FILE;
    }
}

File: src/main/kotlin/com/github/rogerbarton/intellijsv/SvFileType.kt
package com.github.rogerbarton.intellijsv

import com.github.rogerbarton.intellijsv.core.SvIcons
import com.intellij.openapi.fileTypes.LanguageFileType

class SvFileType : LanguageFileType(SvLanguage)
{
    override fun getName() = "System Verilog"

    override fun getIcon() = SvIcons.SV_FILE

    override fun getDefaultExtension() = "sv"

    override fun getDescription() = "System Verilog HDL File"

    companion object {
        val INSTANCE: SvFileType = SvFileType()
    }
}